database-commons/src/main/java/io/cdap/plugin/db/sink/AbstractDBSink.java

```diff
@@ -302,33 +309,15 @@ private Schema inferSchema(Class<? extends Driver> driverClass) {
                                           dbSinkConfig.getJdbcPluginName());
       Properties connectionProperties = new Properties();
       connectionProperties.putAll(dbSinkConfig.getConnectionArguments());
-      try (Connection connection = DriverManager.getConnection(dbSinkConfig.getConnectionString(),
-                                                               connectionProperties)) {
+      try (Connection connection = RetryUtils.createConnectionWithRetry((RetryPolicy<Connection>) retryPolicy,
+        dbSinkConfig.getConnectionString(), connectionProperties, getErrorDetailsProvider())) {
         executeInitQueries(connection, dbSinkConfig.getInitQueries());
-
-        try (Statement statement = connection.createStatement();
-             ResultSet rs = statement.executeQuery("SELECT * FROM " + fullyQualifiedTableName
-                                                     + " WHERE 1 = 0")) {
+        try (Statement statement = RetryUtils.createStatementWithRetry((RetryPolicy<Statement>) retryPolicy,
+          connection, getErrorDetailsProvider());
+             ResultSet rs = RetryUtils.executeQueryWithRetry((RetryPolicy<ResultSet>) retryPolicy, statement,
+               String.format("SELECT * FROM %s WHERE 1 = 0", fullyQualifiedTableName), getErrorDetailsProvider())) {
           inferredFields.addAll(getSchemaReader().getSchemaFields(rs));
         }
-      } catch (SQLException e) {
```

Comment:
we should also add it here then?

Otherwise it will fallback to lower one which will show incorrect error message

Reply:
Added c4469b210f499a1f8cdddb51af87a8e7aa9412dd